sycl/source/detail/graph_impl.hpp

```diff
@@ -383,6 +383,24 @@ class node_impl {
     }
   }
 
+  /// Test if the node contains a N-D copy
+  /// @return true if the op is a N-D copy
+  bool isNDCopyNode() const {
+    if ((MCGType == sycl::detail::CG::CGTYPE::CopyAccToAcc) ||
+        (MCGType == sycl::detail::CG::CGTYPE::CopyAccToPtr) ||
+        (MCGType == sycl::detail::CG::CGTYPE::CopyPtrToAcc)) {
+      sycl::detail::CGCopy *Copy = (sycl::detail::CGCopy *)MCommandGroup.get();
+      sycl::detail::Requirement *ReqSrc =
+          (sycl::detail::Requirement *)(Copy->getSrc());
+      sycl::detail::Requirement *ReqDst =
+          (sycl::detail::Requirement *)(Copy->getDst());
+      if ((ReqSrc->MDims > 1) || (ReqDst->MDims > 1)) {
+        return true;
+      }
+    }
+    return false;
```

Comment:
Super nit, but I personally prefer saving on indentation when possible:
```suggestion
    if ((MCGType != sycl::detail::CG::CGTYPE::CopyAccToAcc) &&
        (MCGType != sycl::detail::CG::CGTYPE::CopyAccToPtr) &&
        (MCGType != sycl::detail::CG::CGTYPE::CopyPtrToAcc))
      return false;
    sycl::detail::CGCopy *Copy = (sycl::detail::CGCopy *)MCommandGroup.get();
    sycl::detail::Requirement *ReqSrc =
        (sycl::detail::Requirement *)(Copy->getSrc());
    sycl::detail::Requirement *ReqDst =
        (sycl::detail::Requirement *)(Copy->getDst());
    return (ReqSrc->MDims > 1) || (ReqDst->MDims > 1);
```
Also, there's something to be said about the use of C-style casts.